igcommit/file_checks.py

```diff
@@ -248,3 +260,83 @@ def __str__(self):
         return '{} "{}" on {}'.format(
             type(self).__name__, self.args[0], self.committed_file
         )
+
+
+class FormatCheck(CommittedFileByExtensionCheck):
+    load_func = None
+    exception_cls = None
+
+    def prepare(self, obj):
+        new = super(FormatCheck, self).prepare(obj)
+        if new and not new.load_func:
+            if not new.configure():
+                return None
+        return new
+
+    def get_problems(self):
+        assert self.load_func and self.exception_cls
+
+        content_proc = self.committed_file.get_content_proc()
+        try:
+            self.load_func(content_proc.stdout.read().decode())
+        except self.exception_cls as error:
+            yield Severity.ERROR, str(error)
+        finally:
+            content_proc.stdout.close()   # Allow it to receive a SIGPIPE
```

Comment:
Ah, I see, didn't dig into the code that much, but maybe it is possible to make `content_proc` to support `with` statements in a way that it will close `stdout` then?